SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp

```diff
@@ -199,162 +159,178 @@ bool FFindWorkerResponseOfType::Update()
 	}
 }
 
-DEFINE_LATENT_AUTOMATION_COMMAND_ONE_PARAMETER(FCleanupConnection, USpatialWorkerConnection*, Connection);
-bool FCleanupConnection::Update()
+struct FConnectionsFixture;
+DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FSetupConnectionFixture, USpatialWorkerConnection*, ServerConnection, USpatialWorkerConnection*, ClientConnection);
+DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FCleanupConnectionFixture, USpatialWorkerConnection*, ServerConnection, USpatialWorkerConnection*, ClientConnection);
+
+struct FConnectionsFixture
 {
-	Connection->RemoveFromRoot();
+	FConnectionsFixture()
+		: ClientConnection(NewObject<USpatialWorkerConnection>())
+		, ServerConnection(NewObject<USpatialWorkerConnection>())
+	{
+		
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupConnectionFixture(ServerConnection, ClientConnection));
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupWorkerConnection(ClientConnection, true));
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupWorkerConnection(ServerConnection, false));
+		ADD_LATENT_AUTOMATION_COMMAND(FWaitForClientAndServerWorkerConnection());
+	}
+
+	~FConnectionsFixture()
+	{
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanupConnectionFixture(ServerConnection, ClientConnection));
+	}
+
+	USpatialWorkerConnection* ClientConnection = nullptr;
+	USpatialWorkerConnection* ServerConnection = nullptr;
+};
+
+bool FSetupConnectionFixture::Update()
+{
+	ClientConnection->AddToRoot();
```

Comment:
Does this need to be in a latent command? I think we could do it directly in the ConnectionFixture constructor. I think there is a very small chance the ServerConnection and ClientConnection could be deallocated in the meantime.

Reply:
It should be fine to have it in constructor instead 👍

Reply:
Addressed.